plc4go/internal/opcua/SecureChannel.go

```diff
@@ -234,15 +233,15 @@ func (s *SecureChannel) submit(ctx context.Context, codec *MessageCodec, errorDi
 					s.log.Debug().Type("type", message).Msg("Not relevant")
 					return false
 				}
-				if requestId := opcuaResponse.GetRequestId(); requestId != transactionId {
+				if requestId := opcuaResponse.GetMessage().GetSequenceHeader().GetRequestId(); requestId != transactionId {
 					s.log.Debug().Int32("requestId", requestId).Int32("transactionId", transactionId).Msg("Not relevant")
 					return false
 				} else {
-					messageBuffer = opcuaResponse.GetMessage()
-					if !(s.senderSequenceNumber.Add(1) == (opcuaResponse.GetSequenceNumber())) {
+					messageBuffer = opcuaResponse.(readWriteModel.BinaryPayload).GetPayload()
```

Comment:
is it ensured that this is always `readWriteModel.BinaryPayloadExactly`?. Otherwise this part will panic